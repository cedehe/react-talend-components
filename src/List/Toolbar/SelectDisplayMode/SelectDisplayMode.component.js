import React, { PropTypes } from 'react';
import { Nav, NavDropdown, MenuItem } from 'react-bootstrap';
import Icon from '../../../Icon';

function getIcon(selected) {
	switch (selected) {
	case 'table': return 'fa-bars';
	case 'large': return 'fa-list-alt';
	case 'tile': return 'fa-th-large';
	default: return 'fa-bars';
	}
}

function getLabel(selected) {
	switch (selected) {
	case 'table': return 'Table';
	case 'large': return 'Expanded';
	case 'tile': return 'Tile';
	default: return 'Table';
	}
}

const displayModes = ['table', 'large', 'tile'];

function SelectDisplayMode({ displayMode, onSelectDisplayMode }) {
	const selected = displayMode || 'table';
	const displayIcon = (<Icon name={getIcon(selected)} />);
	const onSelectMode = (value, e) => {
		onSelectDisplayMode(e, value);
	};
	return (
		<Nav>
			<label className="navbar-text" htmlFor="tc-list-toolbar-display-mode">Display:</label>
			<NavDropdown title={displayIcon} id="tc-list-toolbar-display-mode" onSelect={onSelectMode}>
				{displayModes.map(mode => (
					<MenuItem key={mode} eventKey={mode}>
						<Icon name={getIcon(mode)} /> {getLabel(mode)}
					</MenuItem>
				))}
			</NavDropdown>
		</Nav>
	);
}
SelectDisplayMode.propTypes = {
	onSelectDisplayMode: PropTypes.func,
	displayMode: PropTypes.string,
};

export default SelectDisplayMode;
