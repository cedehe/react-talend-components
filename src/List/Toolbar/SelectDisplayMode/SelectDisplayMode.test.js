import React from 'react';
import renderer from 'react-test-renderer';

import SelectDisplayMode from './SelectDisplayMode.component';

jest.mock('react-dom');

describe('SelectDisplayMode', () => {
	it('should render', () => {
		// given
		const props = {
			onSelectDisplayMode: jest.fn(),
		};

		// when
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render with displayMode = table', () => {
		const props = {
			onSelectDisplayMode: jest.fn(),
			displayMode: 'table',
		};
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with displayMode = large', () => {
		// given
		const props = {
			onSelectDisplayMode: jest.fn(),
			displayMode: 'large',
		};

		// when
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render with displayMode = tile', () => {
		// given
		const props = {
			onSelectDisplayMode: jest.fn(),
			displayMode: 'tile',
		};

		// when
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});


	it('should render id if provided', () => {
		// given
		const props = {
			id: 'toolbar',
			onSelectDisplayMode: jest.fn(),
		};

		// when
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});
});
