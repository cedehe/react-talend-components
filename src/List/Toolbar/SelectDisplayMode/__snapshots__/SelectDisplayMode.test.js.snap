exports[`SelectDisplayMode should render 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-bars"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render id if provided 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor="toolbar-display-mode"
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="toolbar-display-mode"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-bars"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="toolbar-display-mode"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id="toolbar-display-mode-table"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id="toolbar-display-mode-large"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id="toolbar-display-mode-tile"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with displayMode = large 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-list-alt"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with displayMode = table 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-bars"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with displayMode = tile 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-th-large"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;
